### PR TITLE
uast: added augmented assignment roles

### PR DESCRIPTION
**Pull request by juanjux**
[+=, -+, etc](https://en.wikipedia.org/wiki/Augmented_assignment). It like an assignment with an additional operator role that must also be set (or have a child of) one of the operators.

### CODE REVIEW

File: uast/uast.go [hunk 1]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// Augmented assignment: +=, -+, *=, etc
```
Comment (smola) [inline uast/uast.go]: Try to follow the style of other comments. For example, in th is case it wold be something like this:

```
// AugmentedAssignment is an augmented assignment (e.g. +=, -=, *=).
// It is expected that children contains, at least, an AugmentedAssignmentOperator, an AugmentedAssignmentVariable
// and AugmentedAssignmentValue.
AugmentedAssignment
```

Comment (juanjux) [inline uast/uast.go]: Ok!

File: uast/uast.go [hunk 2]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// AugmentedAssignment is an augmented assigment usually combining the equal operator with
+	// another one (e. g. +=, -+, *=, etc). It is expected that childrens containst and
+	// AugmentedAssignmentOperator with a child or aditional role for the specific Bitwise or
+	// Aritmetic operator used. The AugmentedAssignmentVariable and AugmentedAssignmentValue roles
+	// have the same meaning than in Assigment
```
Comment (smola) [inline uast/uast.go]: s/Assigment/Assignment

File: uast/uast.go [hunk 3]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// AugmentedAssignment is an augmented assigment usually combining the equal operator with
+	// another one (e. g. +=, -+, *=, etc). It is expected that childrens containst and
+	// AugmentedAssignmentOperator with a child or aditional role for the specific Bitwise or
+	// Aritmetic operator used. The AugmentedAssignmentVariable and AugmentedAssignmentValue roles
```
Comment (smola) [inline uast/uast.go]: s/Aritmetic/Arithmetic

File: uast/uast.go [hunk 4]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// AugmentedAssignment is an augmented assigment usually combining the equal operator with
```
Comment (smola) [inline uast/uast.go]: s/assigment/assignment

File: uast/uast.go [hunk 5]
```diff
@@ -220,6 +220,16 @@ const (
...
+	// another one (e. g. +=, -+, *=, etc). It is expected that childrens containst and
```
Comment (smola) [inline uast/uast.go]: s/-+/-=

Comment (smola) [inline uast/uast.go]: s/childrens/children
s/containst/contain

File: uast/uast.go [hunk 6]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// AugmentedAssignment is an augmented assignment usually combining the equal operator with
+	// another one (e. g. +=, -=, *=, etc). It is expected that children contains and
+	// AugmentedAssignmentOperator with a child or aditional role for the specific Bitwise or
+	// Arithmetic operator used. The AugmentedAssignmentVariable and AugmentedAssignmentValue roles
+	// have the same meaning than in Assignment
```
Comment (alcortesm) [inline uast/uast.go]: End the sentence with a full stop.

File: uast/uast.go [hunk 7]
```diff
@@ -220,6 +220,16 @@ const (
 	AssignmentVariable
 	AssignmentValue
 
+	// AugmentedAssignment is an augmented assignment usually combining the equal operator with
+	// another one (e. g. +=, -=, *=, etc). It is expected that children contains and
```
Comment (alcortesm) [inline uast/uast.go]: s/and/an/

File: uast/uast.go [hunk 8]
```diff
@@ -220,6 +220,16 @@ const (
...
+	// AugmentedAssignmentOperator with a child or aditional role for the specific Bitwise or
+	// Arithmetic operator used. The AugmentedAssignmentVariable and AugmentedAssignmentValue roles
+	// have the same meaning than in Assignment
+	AugmentedAssignment
+	AugmentedAssignmentOperator
```
Comment (alcortesm) [inline uast/uast.go]: Not sure about this, but... do we really need this constant? we already have the corresponding constants for operators right?

Comment (juanjux) [inline uast/uast.go]: Not really, we have the aritmetic "add" operator (ẗhat is only going to be marked as such on aritmetich expressions on most language AST generators) and the assigment (but not an equal operator, at least not in Python or C, it shows as "comparisonEqual" or "Assigment"). I checked Python, C/C++, Java and D ASTs and either they put all those augmented assignments under the same role with an operator subtoken (D, Python) or they add a different node type for every `x=` operator (C++/Java). ASTs should represent semantic concepts more than tokens.

File: uast/uast.go (hunk 6 above)
Comment (juanjux) [inline uast/uast.go]: Fixed this one and the previous, thanks!

File: uast/uast.go (hunk 8 above)
Comment (alcortesm) [inline uast/uast.go]: :+1:

**Pull request merged**